### PR TITLE
Sample console application

### LINKED ISSUE #614: Prebuilt minimal SVG to PNG converter
Hi! I'm trying to use this library as a simple SVG-to-PNG converter, but I'm not familiar with C#, so I would not be able to build it myself.
Is there are some prebuilt binaries or a build instruction for a complete beginner?

### PR DESCRIPTION
**Pull request by wieslawsoltes**
#### Reference Issue

Fixes #614

#### What does this implement/fix? Explain your changes.

Adds simple command-line application to convert one or many input .svg files to .png images.

```
SvgConsole:
  Converts svg files to encoded png images.

Usage:
  SvgConsole [options]

Options:
  -f, --inputFiles <inputfiles>              The relative or absolute path to the input files
  -d, --inputDirectory <inputdirectory>      The relative or absolute path to the input directory
  -o, --outputDirectory <outputdirectory>    The relative or absolute path to the output directory
  --outputFiles <outputfiles>                The relative or absolute path to the output files
  --width <width>                            The output image width override
  --height <height>                          The output image height override
  --version                                  Show version information
  -?, -h, --help                             Show help and usage information
```

Example usage:

```
./SvgConsole -f ~/svg/Example.svg
```

```
./SvgConsole -f ~/svg/Example.svg --outputFiles ~/png/Example.png
```

```
./SvgConsole -f ~/svg/Example.svg -o ~/png
```

```
./SvgConsole -d ~/svg/
```

```
./SvgConsole -d ~/svg -o ~/png
```

### CODE REVIEW

**Comment by RazrFalcon** [comment]
Could you add an option to render to a specific file and not only into a directory?

**Comment by wieslawsoltes** [comment]
> Could you add an option to render to a specific file and not only into a directory?

Added.

**Comment by H1Gdev** [comment]
Please add this sample app in doc/Samples.md.

**Comment by H1Gdev** [comment]
I have a question.
Why multiple inputs only ?

Can this app be done as follows ?
`./SvgConsole -f a.svg b.svg c.svg --outputFile a.png b.png c.png`

**Comment by wieslawsoltes** [comment]
> I have a question.
> Why multiple inputs only ?
> 
> Can this app be done as follows ?
> `./SvgConsole -f a.svg b.svg c.svg --outputFile a.png b.png c.png`

This can be done too ofcourse. I will add that feature in this PR.

**Comment by wieslawsoltes** [comment]
> Please add this sample app in doc/Samples.md.

Done.

**Comment by wieslawsoltes** [comment]
> I have a question.
> Why multiple inputs only ?
> 
> Can this app be done as follows ?
> `./SvgConsole -f a.svg b.svg c.svg --outputFile a.png b.png c.png`

Added.

**Comment by H1Gdev** [comment]
I think it is better to match these, too.

`--inputDirectories` and `--outputDirectory`.

I think `--inputDirectory` and` --outputDirectory` are good for directory specification.

**Comment by wieslawsoltes** [comment]
> I think it is better to match these, too.(issue is likely to occur with the same file name.)
> 
> `--inputDirectories` and `--outputDirectory`.
> 
> I think `--inputDirectory` and` --outputDirectory` are good for directory specification.

The concept was to have many input files and directories and one output directory. I agree that matching many input files with many output files makes sense, still matching directories is bit trickies, as output directory is used for input files and input directories, so you can do the following:
```
./SvgConsole -f a.svg b.svg c.svg -d ./some/path1/svg ./some/path2/svg --outputDirectory ./some/path/png
```

**Comment by wieslawsoltes** [comment]
> I think it is better to match these, too.(issue is likely to occur with the same file name.)
> 
> `--inputDirectories` and `--outputDirectory`.
> 
> I think `--inputDirectory` and` --outputDirectory` are good for directory specification.

Changed to only allow one input directory.

**Comment by H1Gdev** [comment]
I think that we should as much as possible output files is not overwritten.

> ```
> ./SvgConsole -f a.svg b.svg c.svg -d ./some/path1/svg ./some/path2/svg --outputDirectory ./some/path/png
> ```

In this case, if the following file exists, what is `./some/path/png/a.png` ?
- a.svg
- ./some/path1/svg/a.svg
- ./some/path2/svg/a.svg

**Comment by wieslawsoltes** [comment]
> I think that we should as much as possible output files is not overwritten.
> 
> > ```
> > ./SvgConsole -f a.svg b.svg c.svg -d ./some/path1/svg ./some/path2/svg --outputDirectory ./some/path/png
> > ```
> 
> In this case, if the following file exists, what is `./some/path/png/a.png` ?
> 
> * a.svg
> * ./some/path1/svg/a.svg
> * ./some/path2/svg/a.svg

I think it's up to the user to provide proper input configuration or should we provide more complex logic in sample app? I have changed to allow only one input directory. We can block usage of both input files and input directory together.

File: Samples/SvgConsole/Program.cs [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                }
+            }
+
+            for (int i = 0; i < paths.Count; i++)
+            {
+                var inputPath = paths[i];
+                var outputFile = settings.OutputFiles != null ? settings.OutputFiles[i] : null;
+                try
+                {
+                    string outputPath = string.Empty;
```
Comment (H1Gdev) [inline Samples/SvgConsole/Program.cs]: use `var`.

File: Samples/SvgConsole/Program.cs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+                    if (outputFile != null)
+                    {
+                        outputPath = outputFile.FullName;
+                    }
+                    else
+                    {
+                        var inputExtension = inputPath.Extension;
+                        outputPath = inputPath.FullName.Remove(inputPath.FullName.Length - inputExtension.Length) + ".png";
```
Comment (H1Gdev) [inline Samples/SvgConsole/Program.cs]: better to use `Path.ChangeExtension`.

**Comment by H1Gdev** [comment]
> I think it's up to the user to provide proper input configuration or should we provide more complex logic in sample app? I have changed to allow only one input directory.

Since it is a sample application, it is unclear how much should be allowed, but it is necessary to implement it without misunderstanding because it is a sample application.

File: Samples/SvgConsole/Program.cs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            };
+
+            var optionHeight = new Option(new[] { "--height" }, "The output image height override")
+            {
+                Argument = new Argument<float?>(getDefaultValue: () => null)
+            };
+
+            var rootCommand = new RootCommand()
+            {
+                Description = "Converts a svg file to an encoded png image."
```
Comment (H1Gdev) [inline Samples/SvgConsole/Program.cs]: `Converts svg files to encoded png images.`  :thumbsup:

File: Samples/SvgConsole/Program.cs (hunk 1 above)
Comment (wieslawsoltes) [reply Samples/SvgConsole/Program.cs]: Done.

File: Samples/SvgConsole/Program.cs (hunk 2 above)
Comment (wieslawsoltes) [reply Samples/SvgConsole/Program.cs]: Done.

File: Samples/SvgConsole/Program.cs (hunk 3 above)
Comment (wieslawsoltes) [reply Samples/SvgConsole/Program.cs]: Done.

**Pull request merged**